ambry-router/src/test/java/com.github.ambry.router/MockServer.java

```diff
@@ -53,61 +73,145 @@
    */
   public MockBoundedByteBufferReceive send(Send send)
       throws IOException {
-    RequestOrResponseType type = ((RequestOrResponse) send).getRequestType();
     if (!shouldRespond) {
       return null;
     }
+    ServerErrorCode serverError =
+        hardError != null ? hardError : serverErrors.size() > 0 ? serverErrors.poll() : ServerErrorCode.No_Error;
```

Comment:
this is bit confusing. I guess serverErrors are not used for deletes. Then, why do this. Can we think of moving this inside makePutResponse/makeGetResponse/makeDeleteResponse